### PR TITLE
Throw ObjectDisposeException from HashingStreamImpl when disposed.

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**
This change makes the error that may happen during the "abandoned" copy file more readable.

In some cases, the file copy operation may be canceled from the outside (for instance, by the bandwidth checker). This may cause a different kind of errors because `Stream.CopyTo` procedure may access already disposed streams, and one of the streams may be `HashingStreamImpl` used for on-the-fly file hashing.

Without this change, it was possible to get errors like that:

```
System.AggregateException: One or more errors occurred. ---> System.ObjectDisposedException: Safe handle has been closed
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
   at Microsoft.Win32.Win32Native.SetFilePointerWin32(SafeFileHandle handle, Int32 lo, Int32* hi, Int32 origin)
   at Microsoft.Win32.Win32Native.SetFilePointer(SafeFileHandle handle, Int64 offset, SeekOrigin origin, Int32& hr)
   at System.IO.FileStream.SeekCore(Int64 offset, SeekOrigin origin)
   at System.IO.FileStream.BeginWriteCore(Byte[] bytes, Int32 offset, Int32 numBytes, AsyncCallback userCallback, Object stateObject)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.FileStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at BuildXL.Cache.ContentStore.Hashing.ContentHasher`1.HashingStreamImpl.<WriteAsync>d__38.MoveNext() in \.\Public\Src\Cache\ContentStore\Hashing\ContentHasher.cs:line 410
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.IO.Stream.<CopyToAsyncInternal>d__27.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ObjectDisposedException: Safe handle has been closed
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success)
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success)
   at Microsoft.Win32.Win32Native.SetFilePointerWin32(SafeFileHandle handle, Int32 lo, Int32* hi, Int32 origin)
   at Microsoft.Win32.Win32Native.SetFilePointer(SafeFileHandle handle, Int64 offset, SeekOrigin origin, Int32& hr)
   at System.IO.FileStream.SeekCore(Int64 offset, SeekOrigin origin)
   at System.IO.FileStream.BeginWriteCore(Byte[] bytes, Int32 offset, Int32 numBytes, AsyncCallback userCallback, Object stateObject)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.FileStream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at BuildXL.Cache.ContentStore.Hashing.ContentHasher`1.HashingStreamImpl.<WriteAsync>d__38.MoveNext() in \.\Public\Src\Cache\ContentStore\Hashing\ContentHasher.cs:line 410
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.IO.Stream.<CopyToAsyncInternal>d__27.MoveNext()<---
```
Now, we'll be getting a more graceful error from the hasher without trying to access a closed file stream.

### CODE REVIEW

File: Public/Src/Cache/ContentStore/Hashing/ContentHasher.cs [hunk 1]
```diff
@@ -183,12 +179,13 @@ private sealed class HashingStreamImpl : HashingStream
             private static readonly byte[] EmptyByteArray = new byte[0];
             private static readonly Task<bool> TrueTask = Task.FromResult(true);
 
-            private static readonly Stopwatch _sw = Stopwatch.StartNew();
+            private static readonly Stopwatch Stopwatch = Stopwatch.StartNew();
```
Comment (dcwuser) [inline Public/Src/Cache/ContentStore/Hashing/ContentHasher.cs]: Please avoid instance names that are identical (here even identical casing) to type names. (This is one argument for not using PascalCase for field names.) Maybe timer?

Comment (SergeyTeplyakov) [reply Public/Src/Cache/ContentStore/Hashing/ContentHasher.cs]: Agree. Renamed to Timer.

**Pull request merged**